### PR TITLE
.travis.yml: Use xs-opam repo over deprecated opam-repo-dev

### PR DESCRIPTION
**Pull request by Frezzle**
Signed-off-by: Frederico Mazzone <frederico.mazzone@citrix.com>

### CODE REVIEW

**Comment by Frezzle** [comment]
My branch name of 'fix-travis-build' is misleading(wrong); this gets the build further along, but still fails for another reason - nevertheless, we shouldn't be using opam-repo-dev anymore.

File: .travis.yml [hunk 1]
```diff
@@ -3,4 +3,4 @@ install: wget https://raw.githubusercontent.com/ocaml/ocaml-travisci-skeleton/ma
 script: bash -ex .travis-opam.sh
 sudo: true
 env:
-        - OCAML_VERSION=4.02 PACKAGE=xapi-idl EXTRA_REMOTES=git://github.com/xapi-project/opam-repo-dev
+        - OCAML_VERSION=4.02 PACKAGE=xapi-idl EXTRA_REMOTES=git://github.com/xapi-project/xs-opam
```
Comment (mseri) [inline .travis.yml]: It is picking up cohttp 0.22.0 from some upstream repository, while xs-opam providex 0.21.0 and uchar. I believe Uchar became part of ocaml stdlib in a more recent release.

Can we make sure that the travis opam does not use any upstream repository other than the one in xs-opam?

Comment (mseri) [inline .travis.yml]: Looking at https://github.com/ocaml/ocaml-ci-scripts/blob/master/.travis-ocaml.sh I think we should use `BASE_REMOTE` in place of `EXTRA_REMOTES`

**Comment by Frezzle** [comment]
Even with using BASE_REMOTE over EXTRA_REMOTES it does not work - cohttp is no longer the problem; now it is 'alcotest'.

**Comment by mseri** [comment]
We are fixing it. Hold on :)

**Comment by mseri** [comment]
I am re-running the tests :crossed_fingers:

**Comment by mseri** [comment]
@jonludlam is it ok if I merge?

**Comment by mseri** [comment]
Fred, can you update the travis file to look like `https://github.com/mseri/message-switch/blob/master/.travis.yml`? So that we get build, coverage, and test with usptream opam

File: .travis.yml [hunk 2]
```diff
@@ -3,4 +3,13 @@ install: wget https://raw.githubusercontent.com/ocaml/ocaml-travisci-skeleton/ma
...
+  global:
+    - PACKAGE=xapi-idl
+    - OCAML_VERSION=4.02
+  matrix:
+    - BASE_REMOTE=git://github.com/xapi-project/xs-opam
+    - EXTRA_REMOTES=git://github.com/xapi-project/xs-opam
+matrix:
+  fast_finish: true
+  allow_failure:
+    - OCAML_VERSION=4.02 PACKAGE=xapi-idl EXTRA_REMOTES=git://github.com/xapi-project/xs-opam
```
Comment (mseri) [inline .travis.yml]: Sorry, there was a typo in the example travis file that I gave you. This line should be: 
```
    - env: OCAML_VERSION=4.02 PACKAGE=xapi-idl EXTRA_REMOTES=git://github.com/xapi-project/xs-opam
```

Comment (Frezzle) [inline .travis.yml]: Changed and re-running jobs (although they passed without it).

**Pull request merged**